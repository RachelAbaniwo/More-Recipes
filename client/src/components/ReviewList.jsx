import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';

/**
 * Displays list of reviews
 *
 * @param {oject} props
 *
 * @returns {object} jsx for recipe
 */
const ReviewList = (props) => {
  const ReviewsArray = props.reviews;
  const ReviewsList = ReviewsArray.map(review => (
    <div
      key={review.id}
      style={{
        fontSize: '18px',
        borderBottom: '1px solid lightgrey',
        marginBottom: 20
      }}
    >
      <p className="review">{review.review}</p>
      <p>
        <i className="fa fa-user-circle-o" aria-hidden="true" />
        &nbsp;<span className="title">{review.User.username}</span>
        <small className="text-sm d-block">{format(review.createdAt, 'hh:mm A, do MMMM, YYYY')}</small>
      </p>
    </div>));
  return (
    <article className="review" style={{ paddingBottom: 30 }}>
      {ReviewsList}
    </article>
  );
};

ReviewList.propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    review: PropTypes.string.isRequired,
    userId: PropTypes.number.isRequired
  }))
};
ReviewList.defaultProps = {
  reviews: []
};

export default ReviewList;

