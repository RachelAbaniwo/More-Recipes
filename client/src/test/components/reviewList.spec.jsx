import React from 'react';
import { mount } from 'enzyme';
import ReviewList from '../../components/ReviewList';

describe('The ReviewsList component', () => {
  const props = {
    reviews: [{
      id: 1,
      review: 'some review',
      userId: 1,
      User: {
        id: 1,
        username: 'test_user'
      }
    }, {
      id: 2,
      review: '',
      userId: 1,
      User: {
        id: 1,
        username: 'test_user'
      }
    }]
  };
  it('should display a list of reviews passed to it', () => {
    const wrapper = mount(<ReviewList {...props} />);
    expect(wrapper.find('div').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
