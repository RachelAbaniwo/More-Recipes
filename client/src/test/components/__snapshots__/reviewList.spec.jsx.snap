// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ReviewsList component should display a list of reviews passed to it 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <ReviewList
    reviews={
      Array [
        Object {
          "User": Object {
            "id": 1,
            "username": "test_user",
          },
          "id": 1,
          "review": "some review",
          "userId": 1,
        },
        Object {
          "User": Object {
            "id": 1,
            "username": "test_user",
          },
          "id": 2,
          "review": "",
          "userId": 1,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "reviews": Array [
        Object {
          "User": Object {
            "id": 1,
            "username": "test_user",
          },
          "id": 1,
          "review": "some review",
          "userId": 1,
        },
        Object {
          "User": Object {
            "id": 1,
            "username": "test_user",
          },
          "id": 2,
          "review": "",
          "userId": 1,
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": <article
        class="review"
        style="padding-bottom: 30px;"
      >
        <div
          id="review-list"
        >
          <p
            class="review"
          >
            some review
          </p>
          <p>
            <i
              aria-hidden="true"
              class="fa fa-user-circle-o"
            />
             
            <span
              class="title"
            >
              test_user
            </span>
            <small
              class="text-sm d-block"
            >
              Invalid Date
            </small>
          </p>
        </div>
        <div
          id="review-list"
        >
          <p
            class="review"
          />
          <p>
            <i
              aria-hidden="true"
              class="fa fa-user-circle-o"
            />
             
            <span
              class="title"
            >
              test_user
            </span>
            <small
              class="text-sm d-block"
            >
              Invalid Date
            </small>
          </p>
        </div>
      </article>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="review-list"
          >
            <p
              className="review"
            >
              some review
            </p>
            <p>
              <i
                aria-hidden="true"
                className="fa fa-user-circle-o"
              />
               
              <span
                className="title"
              >
                test_user
              </span>
              <small
                className="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>,
          <div
            id="review-list"
          >
            <p
              className="review"
            >
              
            </p>
            <p>
              <i
                aria-hidden="true"
                className="fa fa-user-circle-o"
              />
               
              <span
                className="title"
              >
                test_user
              </span>
              <small
                className="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>,
        ],
        "className": "review",
        "style": Object {
          "paddingBottom": 30,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            id="review-list"
          >
            <p
              class="review"
            >
              some review
            </p>
            <p>
              <i
                aria-hidden="true"
                class="fa fa-user-circle-o"
              />
               
              <span
                class="title"
              >
                test_user
              </span>
              <small
                class="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="review"
              >
                some review
              </p>,
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle-o"
                />
                 
                <span
                  className="title"
                >
                  test_user
                </span>
                <small
                  className="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>,
            ],
            "id": "review-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p
                class="review"
              >
                some review
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some review",
                "className": "review",
              },
              "ref": null,
              "rendered": Array [
                "some review",
              ],
              "type": "p",
            },
            Object {
              "instance": <p>
                <i
                  aria-hidden="true"
                  class="fa fa-user-circle-o"
                />
                 
                <span
                  class="title"
                >
                  test_user
                </span>
                <small
                  class="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle-o"
                  />,
                  " ",
                  <span
                    className="title"
                  >
                    test_user
                  </span>,
                  <small
                    className="text-sm d-block"
                  >
                    Invalid Date
                  </small>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <i
                    aria-hidden="true"
                    class="fa fa-user-circle-o"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-user-circle-o",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "i",
                },
                " ",
                Object {
                  "instance": <span
                    class="title"
                  >
                    test_user
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test_user",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "test_user",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <small
                    class="text-sm d-block"
                  >
                    Invalid Date
                  </small>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Invalid Date",
                    "className": "text-sm d-block",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Invalid Date",
                  ],
                  "type": "small",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            id="review-list"
          >
            <p
              class="review"
            />
            <p>
              <i
                aria-hidden="true"
                class="fa fa-user-circle-o"
              />
               
              <span
                class="title"
              >
                test_user
              </span>
              <small
                class="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="review"
              >
                
              </p>,
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle-o"
                />
                 
                <span
                  className="title"
                >
                  test_user
                </span>
                <small
                  className="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>,
            ],
            "id": "review-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p
                class="review"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "review",
              },
              "ref": null,
              "rendered": Array [
                "",
              ],
              "type": "p",
            },
            Object {
              "instance": <p>
                <i
                  aria-hidden="true"
                  class="fa fa-user-circle-o"
                />
                 
                <span
                  class="title"
                >
                  test_user
                </span>
                <small
                  class="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle-o"
                  />,
                  " ",
                  <span
                    className="title"
                  >
                    test_user
                  </span>,
                  <small
                    className="text-sm d-block"
                  >
                    Invalid Date
                  </small>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <i
                    aria-hidden="true"
                    class="fa fa-user-circle-o"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-user-circle-o",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "i",
                },
                " ",
                Object {
                  "instance": <span
                    class="title"
                  >
                    test_user
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test_user",
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": Array [
                    "test_user",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": <small
                    class="text-sm d-block"
                  >
                    Invalid Date
                  </small>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Invalid Date",
                    "className": "text-sm d-block",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Invalid Date",
                  ],
                  "type": "small",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "reviews": Array [
          Object {
            "User": Object {
              "id": 1,
              "username": "test_user",
            },
            "id": 1,
            "review": "some review",
            "userId": 1,
          },
          Object {
            "User": Object {
              "id": 1,
              "username": "test_user",
            },
            "id": 2,
            "review": "",
            "userId": 1,
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": <article
          class="review"
          style="padding-bottom: 30px;"
        >
          <div
            id="review-list"
          >
            <p
              class="review"
            >
              some review
            </p>
            <p>
              <i
                aria-hidden="true"
                class="fa fa-user-circle-o"
              />
               
              <span
                class="title"
              >
                test_user
              </span>
              <small
                class="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>
          <div
            id="review-list"
          >
            <p
              class="review"
            />
            <p>
              <i
                aria-hidden="true"
                class="fa fa-user-circle-o"
              />
               
              <span
                class="title"
              >
                test_user
              </span>
              <small
                class="text-sm d-block"
              >
                Invalid Date
              </small>
            </p>
          </div>
        </article>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="review-list"
            >
              <p
                className="review"
              >
                some review
              </p>
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle-o"
                />
                 
                <span
                  className="title"
                >
                  test_user
                </span>
                <small
                  className="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>
            </div>,
            <div
              id="review-list"
            >
              <p
                className="review"
              >
                
              </p>
              <p>
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle-o"
                />
                 
                <span
                  className="title"
                >
                  test_user
                </span>
                <small
                  className="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>
            </div>,
          ],
          "className": "review",
          "style": Object {
            "paddingBottom": 30,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              id="review-list"
            >
              <p
                class="review"
              >
                some review
              </p>
              <p>
                <i
                  aria-hidden="true"
                  class="fa fa-user-circle-o"
                />
                 
                <span
                  class="title"
                >
                  test_user
                </span>
                <small
                  class="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>
            </div>,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="review"
                >
                  some review
                </p>,
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle-o"
                  />
                   
                  <span
                    className="title"
                  >
                    test_user
                  </span>
                  <small
                    className="text-sm d-block"
                  >
                    Invalid Date
                  </small>
                </p>,
              ],
              "id": "review-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <p
                  class="review"
                >
                  some review
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "some review",
                  "className": "review",
                },
                "ref": null,
                "rendered": Array [
                  "some review",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  <i
                    aria-hidden="true"
                    class="fa fa-user-circle-o"
                  />
                   
                  <span
                    class="title"
                  >
                    test_user
                  </span>
                  <small
                    class="text-sm d-block"
                  >
                    Invalid Date
                  </small>
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-user-circle-o"
                    />,
                    " ",
                    <span
                      className="title"
                    >
                      test_user
                    </span>,
                    <small
                      className="text-sm d-block"
                    >
                      Invalid Date
                    </small>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <i
                      aria-hidden="true"
                      class="fa fa-user-circle-o"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-user-circle-o",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "i",
                  },
                  " ",
                  Object {
                    "instance": <span
                      class="title"
                    >
                      test_user
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "test_user",
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "test_user",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <small
                      class="text-sm d-block"
                    >
                      Invalid Date
                    </small>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Invalid Date",
                      "className": "text-sm d-block",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Invalid Date",
                    ],
                    "type": "small",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              id="review-list"
            >
              <p
                class="review"
              />
              <p>
                <i
                  aria-hidden="true"
                  class="fa fa-user-circle-o"
                />
                 
                <span
                  class="title"
                >
                  test_user
                </span>
                <small
                  class="text-sm d-block"
                >
                  Invalid Date
                </small>
              </p>
            </div>,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="review"
                >
                  
                </p>,
                <p>
                  <i
                    aria-hidden="true"
                    className="fa fa-user-circle-o"
                  />
                   
                  <span
                    className="title"
                  >
                    test_user
                  </span>
                  <small
                    className="text-sm d-block"
                  >
                    Invalid Date
                  </small>
                </p>,
              ],
              "id": "review-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <p
                  class="review"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "review",
                },
                "ref": null,
                "rendered": Array [
                  "",
                ],
                "type": "p",
              },
              Object {
                "instance": <p>
                  <i
                    aria-hidden="true"
                    class="fa fa-user-circle-o"
                  />
                   
                  <span
                    class="title"
                  >
                    test_user
                  </span>
                  <small
                    class="text-sm d-block"
                  >
                    Invalid Date
                  </small>
                </p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-user-circle-o"
                    />,
                    " ",
                    <span
                      className="title"
                    >
                      test_user
                    </span>,
                    <small
                      className="text-sm d-block"
                    >
                      Invalid Date
                    </small>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <i
                      aria-hidden="true"
                      class="fa fa-user-circle-o"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-user-circle-o",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "i",
                  },
                  " ",
                  Object {
                    "instance": <span
                      class="title"
                    >
                      test_user
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "test_user",
                      "className": "title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "test_user",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": <small
                      class="text-sm d-block"
                    >
                      Invalid Date
                    </small>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Invalid Date",
                      "className": "text-sm d-block",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Invalid Date",
                    ],
                    "type": "small",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
